src-test/org/etools/j1939_84/controllers/part1/Step05ControllerTest.java

```diff
@@ -0,0 +1,321 @@
+
+        Collection<OBDModuleInformation> obdModuleInformations = new ArrayList<>();
+        OBDModuleInformation obdModule1 = new OBDModuleInformation(0);
+        OBDModuleInformation obdModule2 = new OBDModuleInformation(0);
+        obdModuleInformations.add(obdModule1);
+        obdModuleInformations.add(obdModule2);
+
+        Set<Integer> obdModuleAddresses = new HashSet<>();
+        obdModuleAddresses.add(1);
+        obdModuleAddresses.add(2);
+        obdModuleAddresses.add(3);
+
+        VehicleInformation vehicleInformation = new VehicleInformation();
+        vehicleInformation.setVin("78654321345667889");
+        vehicleInformation.setVehicleModelYear(2006);
+
+        when(vehicleInformationModule.reportVin(any())).thenReturn(packets);
+        when(dataRepository.getObdModuleAddresses()).thenReturn(obdModuleAddresses);
+        when(dataRepository.getVehicleInformation()).thenReturn(vehicleInformation);
+
+        runTest();
+
+        verify(dataRepository, atLeastOnce()).getObdModuleAddresses();
+        verify(dataRepository, atLeastOnce()).getVehicleInformation();
+
+        verify(engineSpeedModule).setJ1939(j1939);
+
+        verify(mockListener).addOutcome(1,
+                5,
+                Outcome.FAIL,
+                "6.1.5.2.b - More than one OBD ECU responded with VIN");
+        verify(mockListener).addOutcome(1,
+                5,
+                Outcome.FAIL,
+                "6.1.5.2.c - VIN does not match user entered VIN");
+        verify(mockListener).addOutcome(1,
+                5,
+                Outcome.FAIL,
+                "6.1.5.2.d - VIN Model Year does not match user entered Vehicle Model Year");
+        verify(mockListener).addOutcome(1,
+                5,
+                Outcome.FAIL,
+                "6.1.5.2.e - VIN is not valid (not 17 legal chars, incorrect checksum, or non-numeric sequence");
+        verify(mockListener).addOutcome(1,
+                5,
+                Outcome.WARN,
+                "6.1.5.3.a - Non-OBD ECU responded with VIN");
+        verify(mockListener).addOutcome(1,
+                5,
+                Outcome.WARN,
+                "6.1.5.3.b - More than one VIN response from an ECU");
+
+        verify(mockListener).addOutcome(1,
+                5,
+                Outcome.WARN,
+                "6.1.5.3.d - Manufacturer defined data follows the VIN");
+
+        verify(vehicleInformationModule).setJ1939(j1939);
+        verify(vehicleInformationModule).reportVin(any());
+
+        verify(vinDecoder).getModelYear("vin");
+        verify(vinDecoder).isVinValid("vin");
+    }
+
+    @Test
+    public void testGetDisplayName() {
+        assertEquals("Display Name", "Part 1 Step 5", instance.getDisplayName());
+    }
+
+    @Test
+    public void testGetTotalSteps() {
+        assertEquals("Total Steps", 1, instance.getTotalSteps());
+    }
+
+    @Test
+    // This test will have an error on the nonObdResponses due to error combination
```

Comment:
It should be possible to run this as the happy path without any warnings or failures recorded.